File: src/main/java/com/Spring/MyPostBank/Models/BankAccount.java
package com.Spring.MyPostBank.Models;

import com.Spring.MyPostBank.Enums.AccountStatus;
import com.Spring.MyPostBank.Enums.AccountType;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@SuperBuilder
public class BankAccount extends AbstractEntity{

    @Column(nullable = false, unique = true)
    private String accountNo;

    @Enumerated(EnumType.STRING)
    private AccountType accountType;

    private BigDecimal balance;

    @Enumerated(EnumType.STRING)
    private AccountStatus status;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "account", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Card> cards;

    @OneToMany(mappedBy = "account", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    private List<Credit> loans;

}

File: mypostbank-ui/src/app/services/models/bank-account.ts
/* tslint:disable */
/* eslint-disable */
import { Card } from '../models/card';
import { Credit } from '../models/credit';
import { FileEntity } from '../models/file-entity';
import { User } from '../models/user';
export interface BankAccount {
  accountNo?: string;
  accountType?: 'CHECKING' | 'SAVINGS';
  balance?: number;
  cards?: Array<Card>;
  createdBy?: number;
  createdDate?: string;
  file?: FileEntity;
  id?: number;
  lastModifiedBy?: number;
  lastModifiedDate?: string;
  loans?: Array<Credit>;
  status?: 'ACTIVE' | 'PENDING' | 'INACTIVE' | 'FROZEN';
  user?: User;
}
